hazelcast/src/main/java/com/hazelcast/core/Cluster.java

```diff
@@ -84,6 +84,18 @@
     Member getLocalMember();
 
     /**
+     * Promotes local lite member to data member.
+     * When this method returns both {@link #getLocalMember()} and {@link #getMembers()}
+     * reflects the promotion.
+     *
+     * @throws IllegalStateException if member is not a lite member
+     * @throws IllegalStateException if mastership claim in progress
+     * @throws IllegalArgumentException if local member cannot be identified as a member of the cluster
```

Comment:
I meant to ask about this line - https://github.com/hazelcast/hazelcast/pull/10142/files#diff-49146434024bde944cb0d360926213efR1090. 
It would mean that it applied a new MemberView but it still was a lite member. But looking at the `MembershipManager#promoteToNormalMember` I can see only 2 outcomes - either it is promoted and the new member view is sent OR and exception is thrown (which will occur on the caller).